src/nvim/api/win_config.c

```diff
@@ -848,6 +865,14 @@ static bool parse_float_config(Dict(float_config) *config, FloatConfig *fconfig,
     fconfig->fixed = config->fixed;
   }
 
+  if (HAS_KEY_X(config, hide)) {
+    if (config->hide && new_win) {
```

Comment:
This error should still be removed. As I said, this would be trivial to emulate using a second `nvim_win_set_config` without an intermediate redraw, so forbidding this just obscures what really is going on.